pkg/didcomm/common/service/service.go

```diff
@@ -20,16 +27,61 @@ type DIDComm interface {
 	Event
 }
 
+// Header helper structure which keeps reusable fields
+type Header struct {
+	ID     string           `json:"@id"`
+	Thread decorator.Thread `json:"~thread"`
+	Type   string           `json:"@type,omitempty"`
+}
+
 // DIDCommMsg did comm msg
 type DIDCommMsg struct {
 	// Outbound indicates the direction of this DIDComm message:
 	//   - outgoing (to another agent)
 	//   - incoming (from another agent)
 	Outbound bool
-	Type     string
 	Payload  []byte
 	// TODO : might need refactor as per the issue-226
 	OutboundDestination *Destination
+	Header              *Header
+}
+
+// GetHeader parses and returns Header from payload
+func (m *DIDCommMsg) GetHeader() (*Header, error) {
```

Comment:
accessors should not be prefixed with Get. Nothing is being “gotten”.

Just Header in this case.

Reply:
I tried, but we need this field to be public. So, in that case, we have a Header field and Header function which lead us to conflict

Reply:
You don’t need both exposed. If you want to expose Header in the struct then you don’t need an accessor.